modules/scheduler-impl/src/main/java/org/opencastproject/scheduler/impl/SchedulerServiceImpl.java

```diff
@@ -732,12 +583,16 @@ private void addEventInternal(Date startDateTime, Date endDateTime, String captu
         mediaPackage.setIdentifier(id);
 
         // Update dublincore title and temporal
-        String newTitle = dc.getFirst(DublinCore.PROPERTY_TITLE) + String.format(" %0" + Integer.toString(periods.size()).length() + "d", ++counter);
+        String newTitle = dc.getFirst(DublinCore.PROPERTY_TITLE) + String.format(" %0" + Integer.toString(periods.size()).length() + "d", currentCounter + 1);
         dc.set(DublinCore.PROPERTY_TITLE, newTitle);
         DublinCoreValue eventTime = EncodingSchemeUtils.encodePeriod(new DCMIPeriod(startDate, endDate),
                 Precision.Second);
         dc.set(DublinCore.PROPERTY_TEMPORAL, eventTime);
-        mediaPackage = updateDublincCoreCatalog(mediaPackage, dc);
+        try {
+          mediaPackage = updateDublincCoreCatalog(mediaPackage, dc);
+        } catch (Exception e) {
+          throw new RuntimeException(e);
```

Comment:
We wanted to preserve the previous behavior of this method which was aborting as soon as an exception occurs. So I think in this case, it's fine to "uncheck" the exceptions since we are in a stream (and you cannot throw checked exceptions in streams).

Reply:
Exactly. Therefore, if the answer is that it does not make sense to cast everything to `RuntimeException`, it does not make sense to use streams here.

Reply:
In general I think it doesn't make sense to cast checked Exceptions to RuntimeExceptions since you are losing compile time safety.

However, in this particular case, I think it is justifiable. These RuntimeExceptions are caught outside the forEach and converted to checked SchedulerExceptions again. So from a caller's point of view, the behavior of the method doesn't really change.

Of course I could also fire up a `ThreadPool` and do a lot of foo instead of using a parallel stream, but I think that is more likely to introduce bugs since you have to shutdown the threadpool properly and so on.

Since Opencast has quite an old code base, many checked exceptions are used. Today, you would probably use less checked exceptions because java8's functional interfaces don't go well with checked exceptions.

Reply:
I will use `Misc.chuck` to be compliant with other Opencast components